charts/py-app/templates/httproute.yaml

```
@@ -0,0 +1,71 @@
+{{- if .Values.httproute.enabled -}}
+---
+{{- $fullName := include "py-app.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: {{ $fullName }}
+  namespace: {{ .Release.Namespace }}
+  {{- with .Values.httproute.annotations }}
+  annotations: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  labels: {{- include "py-app.labels" $ | nindent 4 }}
+    {{- with .Values.httproute.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  parentRefs:
+    - name: {{ .Values.httproute.gatewayName }}
+      namespace: {{ .Values.httproute.gatewayNamespace }}
```

I'm not sure that kubernetes will accept empty string here. 

So it's better to wrap it in some kind of with of if.


```suggestion
	{{- with .Values.httproute.gatewayNamespace }}
      namespace: {{ . }}
    {{- end - }}
```

---

Same goes for the second route.